src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Dataflow/DiagnosticUtilities.cs

```diff
@@ -32,7 +32,15 @@ internal static Origin GetMethodParameterFromIndex(MethodDesc method, int parame
 
         internal static string GetParameterNameForErrorMessage(ParameterOrigin origin)
         {
-            return $"#{origin.Index}";
+            return GetParameterNameForErrorMessage(origin.Method, origin.Index);
+        }
+
+        internal static string GetParameterNameForErrorMessage(MethodDesc method, int parameterIndex)
+        {
+            if (method is EcmaMethod ecmaMethod)
```

Comment:
This will make it handle methods on instantiated types or instantiated generic methods nicely too:

```suggestion
            if (method.GetTypicalMethodDefinition() is EcmaMethod ecmaMethod)
```